DataFormats/Detectors/TPC/include/DataFormatsTPC/ZeroSuppression.h

```diff
@@ -44,6 +46,57 @@ struct TPCZSTBHDR {
   GPUd() const unsigned short* rowAddr1() const { return (unsigned short*)((unsigned char*)this + sizeof(*this)); }
 };
 
+
+struct ZeroSuppressedHeader {
+  /// Header for TPC zero suppression format
+  /// nTimeBins = number of Time bins stored in the current 8KB page
+  /// mCRUID = CRU ID, split into two pages, one per endpoint
+  /// mtimeOffset = time bin offset wrt to the heart beat frame
+  uint8_t version = 0;
+  uint8_t nTimeBins = 0;
+  uint16_t mCRUID = 0;
+  uint16_t mtimeOffset = 0;
+  uint16_t nADCsamples = 0;
+};
+
+struct ZeroSuppressedTimeBinHeader {
+  /// Row mask: highest bit to determine endpoint of CRU: 1 upper, 0 lower
+  /// row Addr: pointer to ZerosuppresedSequences of the row
+  uint16_t rowMask = 0;
+  std::vector<uint16_t> rowAddr = {};
+};
+
+struct ZeroSuppressedRow {
+  //8 bit wide array consisting of the sequences of ADC values
+  //first byte: nSequences
+  //next 2x nSequences bytes: pad number of start and end (one byte each)
+  //following bytes: ADC values
+  uint8_t nSequences = 0;
+  std::vector<uint8_t> sequenceStartEnd = {};
+  std::vector<unsigned char> ADCvalues = {};
+};
+
+struct ZeroSuppressedTimeBin {
+  // Struct for the TPC zero suppressed time bins, to be stored into the ZeroSupressedContainer
+  // Header: k*16 bits, k = # rows per Sector / 2, to split it to the CRU endpoints
+  ZeroSuppressedTimeBinHeader hdr;
+  std::vector<ZeroSuppressedRow> rows = {};
+};
+
+
+struct ZeroSuppressedContainer { // Struct for the TPC zero suppressed data format
+                                 // RDH 64 byte
+                                 // 6 byte header for the zero suppressed format ; 8 bit version, 8 bit number of timebins, 16 bit CRU ID, 16 bit time offset
+                                 // Time bin information
+  uint64_t mRDH[8] = {};         //< 8 * 64 bit RDH (raw data header)
+  ZeroSuppressedHeader hdr;
+  std::vector<ZeroSuppressedTimeBin> timeBin = {};
+};
+
+struct ZSSizes{
+  std::array<unsigned int, 20> s = {};
+};
+
```

Comment:
It seems to me that these structures are not actually needed for the rest of the PR. (They are only used in outcommented code and in cases with verbosity >= 2.) Removing them would also remove the need for the GPU-incompatible headers above.

Reply:
Indeed, I used this for the decoding, which is not yet checked in and needs some cleanup as well. I will have a look if i can remove this completely. For now I moved it to a different header.